pkg/registry/core/pod/strategy.go

```diff
@@ -145,6 +145,9 @@ func (podStrategy) CheckGracefulDelete(ctx context.Context, obj runtime.Object,
 		// use the default value if set, or deletes the pod immediately (0)
 		if pod.Spec.TerminationGracePeriodSeconds != nil {
 			period = *pod.Spec.TerminationGracePeriodSeconds
+			if period == 0 {
```

Comment:
```suggestion
			if period < 1 {
```
This value may be negative, maybe it's better to deal with it together

Reply:
Validation should prevent that.

Reply:
But this verification is missing #98866